rust/signed_doc/src/lib.rs

```diff
@@ -204,8 +204,35 @@ impl CatalystSignedDocument {
     /// Returns a signed document `Builder` pre-loaded with the current signed document's
     /// data.
     #[must_use]
-    pub fn into_builder(&self) -> SignaturesBuilder {
-        self.into()
+    pub fn into_builder(&self) -> anyhow::Result<SignaturesBuilder> {
+        self.try_into()
+    }
+
+    /// Returns data which is used in signing: COSE protected header bytes, COSE payload
+    /// bytes.
+    pub(crate) fn metadata_and_content_bytes(&self) -> anyhow::Result<(&[u8], &[u8])> {
```

Comment:
Can ignore this
Would be great if you can add a CDDL of this, would be easier to visualize